capture/project_library/ot_trace_library/trace_library.py

```diff
@@ -118,10 +118,17 @@ def get_traces(self, start: Optional[int] = None,
             The traces from the database.
         """
         self.flush_to_disk()
-        if start and end:
+        if (start is not None) and (end is not None):
+            # SQL ID starts at 1.
+            start = start + 1
```

Comment:
Don't you need to do the same for `end`? Otherwise one trace less than requested is returned, no?

Reply:
The reason for this is the following:
In Python, `array[0:100]` gives you 100 elements from index 0 to 99.
With the new `get_plaintexts/ciphertexts/waves(0, 100)` functions, we want to imitate this behavior.
As the new trace DB index starts at 1 and not 0, we need an offset of 1, so we are adding +1 in the code.
However, as we still only want 100 elements in this example, we do not add this offset to the end. So we get elements 1 to 100.

Reply:
Thanks for explaining, all good then :-)